src/System.Buffers.Experimental/System/Buffers/BufferExtensions.cs

```diff
@@ -253,6 +253,114 @@ public unsafe static int IndexOfVectorized(this ReadOnlySpan<byte> buffer, byte
             return index;
         }
 
+        public static int ValidateCompileTimeEnumeratorUse(ReadOnlySpan<byte> input, int lastOffset, int maxBytesRemaining)
+        {
+            var upperBound = lastOffset + maxBytesRemaining;
+            int headerStart = lastOffset;
+            foreach (var headerEnd in input.MatchIndicies((byte)'\r', lastOffset, Math.Min(input.Length - lastOffset, maxBytesRemaining)))
+            {
+                if (headerEnd == upperBound)
+                {
+                    throw new Exception("Bad...");
+                }
+
+                if (headerEnd == input.Length)
+                {
+                    // Incomplete
+                    return headerStart;
+                }
+
+                if (input[headerEnd + 1] != (byte)'\n')
+                {
+                    throw new Exception("Bad...");
+                }
+
+                if (headerEnd == headerStart)
+                {
+                    // End of headers
+                    return -1;
+                }
+
+                string headerName = null;
+                string headerValue = null;
+                var isFirst = true;
+                foreach (var valueStart in input.MatchIndicies((byte)':', headerStart, headerEnd - 1))
+                {
+                    if (!isFirst)
+                    {
+                        throw new Exception("Bad...");
+                    }
+                    else
+                    {
+                        isFirst = false;
+                        headerName = input.ToAsciiString(headerStart, valueStart - headerStart);
+                        headerValue = input.ToAsciiString(valueStart + 1, headerEnd - 1 - valueStart);
+                    }
+                }
+
+                if (string.IsNullOrEmpty(headerName) || string.IsNullOrEmpty(headerValue))
+                {
+                    throw new Exception("Bad...");
+                }
+
+                // Do something with headers
+
+                headerStart = headerEnd + 1;
+            }
+
+            if (input.Length >= upperBound)
+            {
+                throw new Exception("Bad...");
+            }
+
+            // Incomplete
+            return headerStart;
+        }
+
+        public static string ToAsciiString(this ReadOnlySpan<byte> buffer)
+        {
+            if (buffer.Length == 0)
+            {
+                return string.Empty;
+            }
+
+            return string.Empty;
```

Comment:
No; they are just filler; need to move that and sample function somewhere else. Was mostly for testing the api.